SolutionDir is undefined when building a project

`"/Users/Dmytro/GitHub/CloudDns/CloudDns/CloudDns.csproj" (default target) (1) ->
(ResolveCodeAnalysisRuleSet target) ->
  /Library/Frameworks/Mono.framework/Versions/5.4.1/lib/mono/msbuild/15.0/bin/Microsoft.CSharp.CurrentVersion.targets(141,9): warning MSB3884: Could not find rule set file "*Undefined*packages/CSharp.Styleguide.1.0.11/StyleCop.ruleset". [/Users/Dmytro/GitHub/CloudDns/CloudDns/CloudDns.csproj]`

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,7 +2,7 @@ language: csharp
 mono: latest
 script:
   - VERSION=$(grep -o -m 1 "[0-9]\+.[0-9]\+" CHANGELOG.md).$TRAVIS_BUILD_NUMBER
-  - sed -e 's/#.#.#/'${VERSION}'/g' -i CSharp.Styleguide.targets
+  - sed -e 's/#.#.#/'${VERSION}'/g' -i CSharp.Styleguide.props
   - nuget pack CSharp.Styleguide.nuspec -Version $VERSION
 before_deploy:
   - curl https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -o nuget.exe -s

--- a/CSharp.Styleguide.nuspec
+++ b/CSharp.Styleguide.nuspec
@@ -11,6 +11,6 @@
   </metadata>
   <files>
     <file src="StyleCop.ruleset" target="" />
-    <file src="CSharp.Styleguide.targets" target="build" />
+    <file src="CSharp.Styleguide.props" target="build" />
   </files>
 </package>

--- a/CSharp.Styleguide.props
+++ b/CSharp.Styleguide.props
@@ -1,8 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <RunCodeAnalysis>true</RunCodeAnalysis>
-    <StyleCopTreatErrorsAsWarnings>false</StyleCopTreatErrorsAsWarnings>
+    <SolutionDir Condition="'$(SolutionDir)'==''">$([System.IO.Path]::GetDirectoryName($(MSBuildProjectDirectory)))\</SolutionDir>
     <CodeAnalysisRuleSet>$(SolutionDir)packages\CSharp.Styleguide.#.#.#\StyleCop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 </Project>

